app/components/Cell/Cell.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function Cell(props) {
+  // console.log('props Cell:')
+  // console.log(props)
+
+  const { x, y, } = props.cellData;
+  props = props.props;
+  const {
+    board, colorA, colorB, colorC, startingPoint, endingPoint,
+  } = props.state;
+  const lengthY = board.length;
+  // const styles = {
+  // 	cell: {
+  // 		backgroundColor:
+  // 	}
+  // }
+
+  return (
+    <div
+      className="cellShell"
+      style={ { height: `${40 / lengthY}vh`, width: `${40 / lengthY}vh`, } }
+    >
+      <div
+        className="cell"
+        style={ {
+backgroundColor:
+											(board[y][x].stepVisited >= 0) ? colorC : (board[y][x].allowed ? colorA : colorB),
+
+        } }
+        onClick={ props.handleClickOnBoard.bind(null, x, y) }
+      >
+        {(startingPoint[0] === x) && (startingPoint[1] === y) && <div>start</div>}
```

Comment:
Convert to component

Reply:
been thinking about that one (and the one below) and got stuck.

* should both of them be one component? or two separate components?
* should that new component return and empty string if the specific tile is neither the starting nor the ending point?
* Or perhaps the component here (and below) should be extremely shallow and just return a `<div> start/end </div>` and the whole boolean  part should still remain here?

Reply:
For conditional things, I usually advise using "MaybeFoo" names/components. Not my invention and it comes straight from functional programming. You can do this:
```jsx
const MaybeStartPoint = (/* input */) => input && condition(s) ? <div>start</div> : null
const MaybeEndPoint = (/* input */) => input && condition(s) ? <div>end</div> : null
// then use it:
<MaybeStartPoint {...props} />
<MaybeEndPoint {...props} />
```